### ISSUE TITLE
Python 3.7 support broken

### ISSUE THREAD

**Issue by sachinkamath** [owtf/owtf#1018]
I am using the python 3.7.2 version on Arch Linux and one of the pinned dependencies of OWTF, `pexpect` throws an error.

## Current Behavior

```
$ owtf                                                                                                                                                                           

         _____ _ _ _ _____ _____
        |     | | | |_   _|   __|
        |  |  | | | | | | |   __|
        |_____|_____| |_| |__|

            @owtfp
        http://owtf.org
        Version: 2.6.0
        
[+] Loading /home/neo/.owtf/conf/framework.yaml..
[+] Loading /home/neo/.owtf/conf/general.yaml..
[+] Loading resources from: /home/neo/.owtf/conf/resources.cfg..
Traceback (most recent call last):
  File "/home/neo/.virtualenvs/owtf/bin/owtf", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/neo/projects/owtf/bin/owtf", line 11, in <module>
    main()
  File "/home/neo/projects/owtf/owtf/core.py", line 337, in main
    load_plugins(db)
  File "/home/neo/projects/owtf/owtf/managers/plugin.py", line 145, in load_plugins
    os.path.splitext(file)[0], filename, pathname, desc
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/neo/projects/owtf/owtf/plugins/auxiliary/smb/SMB_Handler@OWTF-SMB-001.py", line 3, in <module>
    from owtf.protocols.smb import SMB
  File "/home/neo/projects/owtf/owtf/protocols/smb.py", line 11, in <module>
    from owtf.shell import pexpect_sh
  File "/home/neo/projects/owtf/owtf/shell/pexpect_sh.py", line 9, in <module>
    import pexpect
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pexpect/__init__.py", line 75, in <module>
    from .pty_spawn import spawn, spawnu
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pexpect/pty_spawn.py", line 14, in <module>
    from .spawnbase import SpawnBase
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pexpect/spawnbase.py", line 224
    def expect(self, pattern, timeout=-1, searchwindowsize=-1, async=False):
                                                                   ^
SyntaxError: invalid syntax
```

## Possible Solution

On some digging, I found that this issue was fixed in the [upstream](https://github.com/pexpect/pexpect/blob/c741080a46445e5b2dec037c4856617f679d4546/pexpect/spawnbase.py) from `async=True` to `async_=True`. I manually updated pexpect to the latest version using pip and got another error from OWTF because of the requirements parser.

```$ owtf                                                                                                                                                                             
Traceback (most recent call last):
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (pexpect 4.6.0 (/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages), Requirement.parse('pexpect==4.2.1'), {'owtf'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/neo/.virtualenvs/owtf/bin/owtf", line 4, in <module>
    __import__('pkg_resources').require('owtf==2.4.0')
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3191, in <module>
    @_call_aside
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3175, in _call_aside
    f(*args, **kwargs)
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3204, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/home/neo/.virtualenvs/owtf/lib/python3.7/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pexpect==4.2.1' distribution was not found and is required by owtf
```

The fix is pretty straight forward. The requirements has to be updated accordingly in the requirements and it should be good to go.

## Steps to Reproduce (for bugs)

1. Install Python 3.7
2. Run setup script
3. Run OWTF

## Your Environment

* Installation method used: Defult setup
* Version/branch used: develop
* Operating System and version (like Kali, Debian, ArchLinux, etc): ArchLinux
* pip/setuptools version: `pip 19.0.3 (python 3.7)`
* sudo access: `True`

**Comment by sachinkamath** [comment]
Can confirm the fix works on Python 3.7

```
$ owtf                                                                                                                                                                           

         _____ _ _ _ _____ _____
        |     | | | |_   _|   __|
        |  |  | | | | | | |   __|
        |_____|_____| |_| |__|

            @owtfp
        http://owtf.org
        Version: 2.6.0
        
[+] Loading /home/neo/.owtf/conf/framework.yaml..
[+] Loading /home/neo/.owtf/conf/general.yaml..
[+] Loading resources from: /home/neo/.owtf/conf/resources.cfg..
[+] Loading framework please wait..
[!] Starting HTTP(s) proxy server at 127.0.0.1:8008
[!] Starting API and UI server at http://0.0.0.0:8009
```

**Comment by viyatb** [comment]
Merged the fix.
